src/main/java/org/jboss/staxmapper/XMLMapperImpl.java

```diff
@@ -100,10 +108,24 @@ public void deparseDocument(XMLContentWriter contentWriter, XMLStreamWriter stre
         contentWriter.writeContent(new FormattingXMLStreamWriter(streamWriter));
     }
 
-    @SuppressWarnings({ "unchecked" })
+    private XMLElementReader<?> getParser(final QName name) {
+        rootElements.computeIfAbsent(name, qName -> {
+            XMLElementReader<?> reader = rootElementsProvider.get(name).get(); //lazy create of parser
+            return new WeakReference<>(reader);
```

Comment:
jira https://issues.jboss.org/browse/WFCORE-1841
Yes that is the idea, that parsers that are no longer needed can be discarded, as they can always be recreated by providers. In usual scenario we have is that parser is only used when booting up the server or re configuring/reloading it but in normal running scenario, server is started and than up for weeks if not for months, so why keep all versions of parsers of all subsystems always in memory if they are not needed.